### PR TITLE
raft: port paused follower resuming on MsgHeartbeatResp

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR tries to fix issue #1426. It also includes the test case fix https://github.com/coreos/etcd/pull/7118.

### CODE REVIEW

**Comment by hhkbp2** [comment]
PTAL @siddontang @BusyJay @ngaut

**Comment by siddontang** [comment]
LGTM

File: tests/test_raft.rs [hunk 1]
```diff
@@ -534,19 +534,25 @@ fn test_progress_resume() {
...
     raft.prs.get_mut(&2).unwrap().paused = true;
-    let mut m = Message::new();
-    m.set_from(1);
-    m.set_to(1);
-    m.set_msg_type(MessageType::MsgBeat);
-    raft.step(m).expect("");
-    assert!(!raft.prs[&2].paused, "paused = true, want false");
+
+    raft.step(new_message(1, 1, MessageType::MsgBeat, 0)).expect("");
+    assert!(raft.prs[&2].paused,
```
Comment (BusyJay) [inline tests/test_raft.rs]: Just `assert!(raft.prs[&2].paused)` is ok.

File: tests/test_raft.rs [hunk 2]
```diff
@@ -534,19 +534,25 @@ fn test_progress_resume() {
...
+            "paused = {}, want true",
+            raft.prs[&2].paused);
+
+    raft.prs.get_mut(&2).unwrap().become_replicate();
+    raft.step(new_message(2, 1, MessageType::MsgHeartbeatResponse, 0)).expect("");
+    assert!(!raft.prs[&2].paused,
```
Comment (BusyJay) [inline tests/test_raft.rs]: Ditto.

File: tests/test_raft.rs [hunk 3]
```diff
@@ -1433,32 +1439,22 @@ fn test_handle_heartbeat_resp() {
...
     sm.step(m).expect("");
     // Consume the message sent in response to MsgAppResp
     sm.read_messages();
 
-    sm.bcast_heartbeat(); // reset wait state
     sm.step(new_message(2, 0, MessageType::MsgHeartbeatResponse, 0)).expect("");
     msgs = sm.read_messages();
-    assert_eq!(msgs.len(), 1);
-    assert_eq!(msgs[0].get_msg_type(), MessageType::MsgHeartbeat);
+    assert_eq!(msgs.len(), 0);
```
Comment (BusyJay) [inline tests/test_raft.rs]: Use `is_empty()` instead.

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
I have approved it 14 minutes ago.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**